packages/terra-avatar/src/Avatar.jsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import TerraImage from 'terra-image';
+import styles from './Avatar.scss';
+
+const cx = classNames.bind(styles);
+
+const AvatarVariants = {
+  FACILITY: 'facility',
+  USER: 'user',
+};
+
+const propTypes = {
+  /**
+   * The text content that specifies the alternative text for the image.
+   */
+  alt: PropTypes.string,
+  /**
+   * The text content of the aria-label for accessibility.
+   */
+  ariaLabel: PropTypes.string,
+  /**
+   * The image to display.
+   */
+  image: PropTypes.string,
+  /**
+   * Two or three letters to display.
+   */
+  initials: PropTypes.string,
+  /**
+   * The avatar variant. One of `AvatarVariants.FACILITY`, `AvatarVariants.USER`.
+   */
+  variant: PropTypes.oneOf([...Object.values(AvatarVariants)]),
+};
+
+const defaultProps = {
+  alt: undefined,
+  ariaLabel: undefined,
+  image: undefined,
+  initials: undefined,
+  variant: AvatarVariants.USER,
+};
+
+const Avatar = ({
+  alt,
+  ariaLabel,
+  image,
+  initials,
+  variant,
+  ...customProps
+  }) => {
+  // Checks to run when not in production
+  if (process.env.NODE_ENV !== 'production') {
+    if (image && initials) {
+      // eslint-disable-next-line
+      console.warn('Only one of the props: [image, initials] should be supplied.');
+    }
+  }
+
+  const attributes = Object.assign({}, customProps);
+
+  const avatarClassNames = cx([
+    'avatar',
+    attributes.className,
+  ]);
+
+  const avatarTextClassNames = cx([
+    { 'avatar-text-small': initials && initials.length === 3 },
+    { 'avatar-text-large': initials && initials.length === 2 },
+  ]);
+
+  const icon = (
+    <div className={cx('avatar-icon')}>
+      <svg className={cx(variant)} />
+    </div>
+  );
+
+  let avatarContent;
+  if (image) {
+    avatarContent = <TerraImage className={cx('avatar-image')} src={image} placeholder={icon} alt={alt} />;
+  } else if (initials && (initials.length === 2 || initials.length === 3)) {
```

Comment:
This seems misleading, I propose we splice any initials provided to the first three letters, otherwise users may not understanding why their prop is not honored.

Reply:
If we only take the first 3 letters of string, we should update the `initials` prop description from `Two or three letters to display.` to describe this behavior.

Reply:
For the case of initials I think splicing may be a little deceiving. Removing the last character(s) representing a last name or facility name could make the initials misleading and potentially un-mappable if there is another valid set of initials that are a subset of the spliced initials.

I can see the confusion of not showing any initials as well, so I guess the question becomes is an incorrect / incomplete set of initials more beneficial than the fallback icon? Is this a situation where we should manipulate user data rather than log a warning and fall back to the icon?

Reply:
I prefer the fallback approach. I don't like the idea of allowing components to be so opinionated that they modify data. The purpose of the fallback icon is to, IMO, is not only to provide a _default_ display for a user, facility, etc. (in the future), but to provide us a safe escape route for a consumer improperly using the component

Reply:
I agree with Dylan, it is better to fallback on the icon incase the required initials are not passed in. Lets go with that.

Reply:
Yeah, I agree with the consensus that we should keep the fallback behavior and not add the splicing behavior.

Reply:
+1 to fallback. I think this fallback strategy should be outline in the props above to clarify the prop priority